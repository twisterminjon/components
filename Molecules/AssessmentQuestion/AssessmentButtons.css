.assessment-buttons {
  display: flex;
  flex-shrink: 0;
  flex-direction: column;
  background: var(--assessment__bg);
  padding: 20px;
  margin-top: 10px;
}

.assessment-buttons-previous {
  margin-top: 10px !important;
}
