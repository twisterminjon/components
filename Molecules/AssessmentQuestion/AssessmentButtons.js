import React from 'react'
import PropTypes from 'prop-types'

import Button from '../../Atoms/Button/Button'
import ButtonGhost from '../../Atoms/ButtonGhost/ButtonGhost'

import './AssessmentButtons.css'

AssessmentButtons.propTypes = {
  /** Executes on `Go back` */
  onPrevious: PropTypes.func.isRequired,

  /** Executes on `Next` */
  onNext: PropTypes.func.isRequired,

  /** Is move forward available */
  allowNext: PropTypes.bool.isRequired,

  /** Loadings status */
  status: PropTypes.shape({
    loadingForward: PropTypes.bool.isRequired,
    loadingBack: PropTypes.bool.isRequired,
  }).isRequired,
}

function AssessmentButtons({ status, allowNext, onPrevious, onNext }) {
  const disabled = status.loadingForward || status.loadingBack
  return (
    <div className="assessment-buttons">
      <Button
        fluid
        disabled={disabled || !allowNext}
        loading={status.loadingForward}
        onClick={onNext}
        data-testid="assessment-question-move-forward">
        Next
      </Button>
      <ButtonGhost
        fluid
        disabled={disabled}
        loading={status.loadingBack}
        className="assessment-buttons-previous"
        onClick={onPrevious}
        data-testid="assessment-question-previous">
        Go back
      </ButtonGhost>
    </div>
  )
}

export default AssessmentButtons
