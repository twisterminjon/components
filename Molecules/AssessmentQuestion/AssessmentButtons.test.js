import React from 'react'

import AssessmentButtons from './AssessmentButtons'

describe('AssessmentButtons', () => {
  const mockFun = jest.fn()

  it('matches the snapshot', () => {
    const wrapper = window.shallow(
      <AssessmentButtons
        allowNext={true}
        onNext={mockFun}
        onPrevious={mockFun}
        status={{
          loadingForward: true,
          loadingBack: false,
        }}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
