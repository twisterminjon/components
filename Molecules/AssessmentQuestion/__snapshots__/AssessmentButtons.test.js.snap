// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssessmentButtons matches the snapshot 1`] = `
<div
  className="assessment-buttons"
>
  <Button
    data-testid="assessment-question-move-forward"
    disabled={true}
    fluid={true}
    loading={true}
    onClick={[MockFunction]}
    style={Object {}}
    type="submit"
  >
    Next
  </Button>
  <ButtonGhost
    className="assessment-buttons-previous"
    data-testid="assessment-question-previous"
    disabled={true}
    fluid={true}
    loading={false}
    onClick={[MockFunction]}
    red={false}
    style={Object {}}
  >
    Go back
  </ButtonGhost>
</div>
`;
